docs/website/scripts/live-blocks/src/preprocess/acquireOPAVersion.js

```diff
@@ -38,11 +45,21 @@ export async function acquireOPAVersion(version) {
 // WARNING: This function and it's resulting acquirer should only get called once per version.
 function createAcquirer(version) {
   return async () => {
+
+    println(`Locating OPA version ${version}...`)
+
     const path = pathToVersion(version)
+
     if (await needsDownloading(version, path)) { // This does not protect against run conditions within a single preprocessing script run, simply allows (long-term) caching.
 
-      println(`Locating OPA version ${version}...`)
-      const assetURL = await getAssetURL(version)
+      // Edge is required to have been built from the current source tree,
+      // expect the binary to be available at the normal build output location.
+      // We will _NOT_ download anything for it.
+      if (version === VERSION_EDGE) {
+        throw new Error(`binary for ${VERSION_EDGE} is not available at path ${path}, ensure it has been built for the current platform`)
```

Comment:
Just curious why this branch is needed. Are we trying to protect against this code getting accidentally called for the edge version?

Reply:
Correct, if `needsDownloading(version, path)` for the edge version comes back as true we need to raise an error. That check basically ensures the path exists and is an executable.